wagtail_extensions/blocks.py

```diff
@@ -81,6 +81,11 @@ def clean(self, value):
             raise ValidationError("There is a problem with this department", params=errors)
         return super().clean(value)
 
+    def to_python(self, value):
+        value = super().to_python(value)
+        value['non_empty_phones'] = [p for p in value.get('phones') if p != '']
```

Comment:
I find this a bit confusing - why should the template expect anything other than non-empty phones? There must be a way to remove empty values before saving the object?

Reply:
@solarissmoke It isn't quite as straight forward as that. The list block intentionally injects an empty child item and this is used when rendering the form etc. We can define a custom `StrippedListBlock` but the only place to do the filtering without messing up form rendering would be in `get_context`, which just effects template rendering.

Reply:
OK. I am fairly sure we will forget about this and still try to access `value.phones` in the template in future projects. In which case maybe it's better just to check the emptiness-or-not in the template anyway?

Reply:
Don't mind leaving `non_empty_phones` there though - just that we're likely to repeat the effort in the template anyway.